Library/Homebrew/cmd/update.sh

```diff
@@ -378,19 +381,34 @@ EOS
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
+    cd "$DIR" || continue
+
+    TAP_VAR="$(repo_var "$DIR")"
+    declare PREFETCH_REVISION"$TAP_VAR"="$(read_current_revision)"
```

Comment:
This line broke `brew update` for anyone who updated between this commit being merged and the fix in 985c672bac4dc20d369b451c484eb6553762dbcf. It read the current revision of that branch rather than the e.g. `origin/master` revision.